### PR TITLE
 fix: replace Integer.parseInt with safe default fallback layout size in 

### LINKED ISSUE #2595: [Bug] EctSaveEncounter2Action: unguarded Integer.parseInt() on layout parameters causes 500 on malformed input (claude assist)
## Summary

`EctSaveEncounter2Action.execute()` calls `Integer.parseInt()` on four request parameters (`rowOneSize`, `rowTwoSize`, `rowThreeSize`, `presBoxSize`) with no prior null check or numeric validation. Any missing or non-numeric value throws an unhandled `NumberFormatException`, resulting in an unhandled 500 error.

## Evidence

File: `src/main/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctSaveEncounter2Action.java:280–283`
\`\`\`java
ew.setRowOneSize(Integer.parseInt(httpservletrequest.getParameter("rowOneSize")));
ew.setRowTwoSize(Integer.parseInt(httpservletrequest.getParameter("rowTwoSize")));
ew.setRowThreeSize(Integer.parseInt(httpservletrequest.getParameter("rowThreeSize")));
ew.setPresBoxSize(Integer.parseInt(httpservletrequest.getParameter("presBoxSize")));
\`\`\`

## Steps to reproduce

Submit the encounter save form with a non-numeric value (or empty value) for any layout parameter.

## Suggested fix

Wrap each parse with a null check or use `ConversionUtils.fromIntString()` (already used elsewhere in the file) with a safe default, e.g.:
\`\`\`java
ew.setRowOneSize(ConversionUtils.fromIntString(httpservletrequest.getParameter("rowOneSize"), 50));
\`\`\`

🤖 Generated with [Claude Code](https://claude.ai/code)

### PR DESCRIPTION
**Pull request by maanvik**
…parsing

## Description

`EctSaveEncounter2Action.execute()` called `Integer.parseInt()` on four request parameters (`rowOneSize`, `rowTwoSize`, `rowThreeSize`, `presBoxSize`) with no null check or validation. Any missing or non-numeric value threw an unhandled `NumberFormatException`, resulting in a 500 error.

Replaced the four `parseInt` calls with a `parseLayoutSize()` helper that uses `ConversionUtils.fromIntString()` for null/non-numeric safety, then clamps any value below the minimum UI height (10px) to the field's known default (60 / 60 / 378 / 30).

## Related Issues

Related to #2595

## How Was This Tested?
Added unit tests in `EctSaveEncounter2ActionTest` covering:
- `null` parameter → returns default
- Non-numeric string → returns default
- Negative value → returns default
- Value below minimum (9) → returns default
- Valid value (50) → passes through unchanged

## Checklist

- [x] My commits are signed off for the [DCO](https://developercertificate.org/) (`git commit -s`)
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/) format, or I've written clear commit messages and will use the format next time
- [x] I have not included any patient data (PHI) in this PR
- [x] I have added tests for new functionality, or this change doesn't need new tests
- [x] I have read the [contributing guide](CONTRIBUTING.md)

## Summary by Sourcery

Handle encounter layout size parameters safely to avoid failures from invalid request values.

Bug Fixes:
- Prevent unhandled errors when encounter layout size request parameters are null, non-numeric, or below the minimum size by falling back to sane defaults.

Tests:
- Add unit tests covering the layout size parsing helper for null, non-numeric, negative, below-minimum, and valid values.

---
## Summary by cubic
Prevent 500s when saving encounter layout by safely parsing layout size params and falling back to defaults. Enforces a 10px minimum; invalid or too-small values no longer crash.

- **Bug Fixes**
  - Replaced `Integer.parseInt` for `rowOneSize`, `rowTwoSize`, `rowThreeSize`, `presBoxSize` with `parseLayoutSize()` using `ConversionUtils.fromIntString`; values <10 or non-numeric/null use defaults (60, 60, 378, 30), 10+ pass; logs a debug message on fallback.
  - Added unit tests in `EctSaveEncounter2ActionUnitTest` for null, non-numeric, negative, below-min, equals-min, and valid inputs.

<sup>Written for commit 76f0cafddb8b3c949ef64e15f1783549c394ed06. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/carlos-emr/carlos/pull/2886?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/test/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctSaveEncounter2ActionTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static io.github.carlos_emr.carlos.encounter.pageUtil.EctSaveEncounter2Action.parseLayoutSize;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Tag("unit")
+@Tag("fast")
+@Tag("encounter")
+class EctSaveEncounter2ActionTest {
```
[the remark this answers is not in the document]
Comment (maanvik) [reply src/test/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctSaveEncounter2ActionTest.java]: Repository now only contains src/test

File: src/test/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctSaveEncounter2ActionUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Test
+    void shouldReturnDefault_whenValueIsBelowMinimum() {
+        assertThat(parseLayoutSize("9", 60)).isEqualTo(60);
+    }
+
+    @Test
+    void shouldReturnValue_whenParamIsValid() {
+        assertThat(parseLayoutSize("50", 60)).isEqualTo(50);
+    }
```
[the remark this answers is not in the document]
Comment (maanvik) [reply src/test/java/io/github/carlos_emr/carlos/encounter/pageUtil/EctSaveEncounter2ActionUnitTest.java:43]: Added most of edge cases, didn't wanted to add too many test cases for testing a simple parse method

**Comment by Ben-Heerema** [comment]
@coderabbitai review
@cubic-dev-ai review

**Comment by Ben-Heerema** [comment]
Hey @maanvik, great work so far, the change looks pretty good to me. 

One thing I’d like changed before merge: please rename `EctSaveEncounter2ActionTest` to `EctSaveEncounter2ActionUnitTest`. The test passes when run explicitly, but this package is not covered by the current Surefire `**/*Test.java` includes. Using the `*UnitTest` suffix ensures the new regression coverage runs under normal `mvn test` / CI discovery.

And while you're touching up that nitpick, seems like coderabbit also has some that may be worth looking into :). 

Thanks for your work thus far! Also if you want to manually call coderabbit you should be able to, just copy what I did above :)

**Comment by maanvik** [comment]
@Ben-Heerema Thanks for the review!
I have made the suggested changes; feel free to review when you get some time.
Thanks!

**Comment by Ben-Heerema** [comment]
@coderabbitai review
@cubic-dev-ai review

**Comment by Ben-Heerema** [comment]
@maanvik this looks awesome! Great work! If there are other tickets you want to work on they're free game! Thanks!

@yingbull ready for review

**Review by yingbull — APPROVED** [review]

**Comment by yingbull** [comment]
@maanvik Appreciate the contribution.  Thanks so much.
